src/storage-client/src/sink.rs

```diff
@@ -234,7 +257,13 @@ pub async fn publish_kafka_schemas(
     Ok((key_schema_id, value_schema_id))
 }
 
-/// Ensures that the Kafka sink's data and consistency collateral exist.
+/// Ensures that the Kafka sink's data and consistency collateral exist, as well
+/// as returning the last complete timestamp that the last incarnation of the
+/// sink committed to Kafka.
+///
+/// IMPORTANT: to achieve exactly once guarantees, the producer that will resume
+/// production at the returned timestamp *must* have called `init_transactions`
+/// prior to calling this method.
```

Comment:
@sploiselle I only changed this because I had written a similar warning in #23438 on `determine_latest_progress_record` and it seemed worth keeping the two warnings in line. I figure the warning isn't going to last long anyway, because I think we've all independently determined it's silly to have these functions in `storage-client`, far away from their single caller in the sink codebase.